feat: add `ndarray/pop`

headlessNode opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves https://github.com/stdlib-js/metr-issue-tracker/issues/78.

## Description

> What is the purpose of this pull request?

This pull request:

-   add `ndarray/pop`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves https://github.com/stdlib-js/metr-issue-tracker/issues/78
-   progresses #2656

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add ndarray/pop

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/ndarray/pop/README.md:
````diff
@@ -0,0 +1,148 @@
+var pop = require( '@stdlib/ndarray/pop' );
+```
+
+#### pop( x, dim )
````
Should we default to `dim = -1` here and make `dim` an optional argument?

kgryte replied:
This would make `pop( vector )` a more direct analog to `Array.prototype.pop`, rather than needing to do `pop( vector, 0 )`, which is more verbose when `dim` can only be `0` (or `-1`).

headlessNode replied:
I also had the same thought when implementing. I'll go ahead and make the change

kgryte commented on lib/node_modules/@stdlib/ndarray/pop/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@stdlib/ndarray/pop",
+  "version": "0.0.0",
+  "description": "Return an array containing a read only truncated view of an input ndarray and a read only view of the last element(s) along a specified dimension.",
```
```suggestion
  "description": "Return an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specified dimension.",
```

headlessNode committed:
fix: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
refactor: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
refactor: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/README.md:
```diff
@@ -0,0 +1,184 @@
+
+# pop
+
+> Return an array containing a read-only truncated view of an input [`ndarray`][@stdlib/ndarray/ctor] and a read-only view of the last element(s) along a specified dimension.
```
```suggestion
> Return an array containing a read-only truncated view of an input [`ndarray`][@stdlib/ndarray/ctor] and a read-only view of the last element(s) along a specific dimension.
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/README.md:
```diff
@@ -0,0 +1,184 @@
+
+#### pop( x\[, options] )
+
+Returns an array containing a **read-only** truncated view of an input [`ndarray`][@stdlib/ndarray/ctor] and a **read-only** view of the last element(s) along a specified dimension.
```
```suggestion
Returns an array containing a **read-only** truncated view of an input [`ndarray`][@stdlib/ndarray/ctor] and a **read-only** view of the last element(s) along a specific dimension.
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/docs/repl.txt:
```diff
@@ -0,0 +1,38 @@
+
+{{alias}}( x[, options] )
+    Returns an array containing a read-only truncated view of an input ndarray
+    and a read-only view of the last element(s) along a specified dimension.
```
```suggestion
    and a read-only view of the last element(s) along a specific dimension.
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/docs/types/index.d.ts:
```diff
@@ -0,0 +1,73 @@
+}
+
+/**
+* Returns an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specified dimension.
```
```suggestion
* Returns an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specific dimension.
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/lib/index.js:
```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+/**
+* Return an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specified dimension.
```
```suggestion
* Return an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specific dimension.
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/lib/main.js:
```diff
@@ -0,0 +1,97 @@
+// MAIN //
+
+/**
+* Returns an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specified dimension.
```
```suggestion
* Returns an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specific dimension.
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@stdlib/ndarray/pop",
+  "version": "0.0.0",
+  "description": "Return an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specified dimension.",
```
```suggestion
  "description": "Return an array containing a read-only truncated view of an input ndarray and a read-only view of the last element(s) along a specific dimension.",
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/lib/main.js:
```diff
@@ -0,0 +1,97 @@
+* @throws {TypeError} first argument must be an ndarray having one or more dimensions
+* @throws {RangeError} dimension index exceeds the number of dimensions
+* @throws {TypeError} first argument must be an ndarray having one or more dimensions
+* @throws {TypeError} second argument must be an integer
```
```suggestion
* @throws {TypeError} options arguments must be an object
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/lib/main.js:
```diff
@@ -0,0 +1,97 @@
+*
+* @param {ndarray} x - input array
+* @param {Object} options - functions options
+* @param {integer} options.dim=-1 - dimension along which to perform the operation
```
```suggestion
* @param {Object} [options] - functions options
* @param {integer} [options.dim=-1] - dimension along which to perform the operation
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/lib/main.js:
```diff
@@ -0,0 +1,97 @@
+* @param {integer} [options.dim=-1] - dimension along which to perform the operation
+* @throws {TypeError} first argument must be an ndarray having one or more dimensions
+* @throws {RangeError} dimension index exceeds the number of dimensions
+* @throws {TypeError} first argument must be an ndarray having one or more dimensions
+* @throws {TypeError} options arguments must be an object
+* @returns {Array<ndarray>} a list of ndarray views
```
```suggestion
* @throws {TypeError} options arguments must be an object
* @throws {TypeError} must provide valid options
* @returns {Array<ndarray>} a list of ndarray views
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode commented on lib/node_modules/@stdlib/ndarray/pop/README.md:
```diff
@@ -0,0 +1,184 @@
+
+-   **dim**: dimension along which to perform the operation. If provided an integer less than zero, the dimension index is resolved relative to the last dimension, with the last dimension corresponding to the value `-1`. Default: `-1`.
+
+By default, the function performs the operation along the last dimension of the input [ndarray][@stdlib/ndarray/ctor]. To perform the operation on a desired dimension, specify the `dim` option.
```
```suggestion
By default, the function performs the operation along the last dimension of the input [ndarray][@stdlib/ndarray/ctor]. To perform the operation along a desired dimension, specify the `dim` option.
```

headlessNode committed:
docs: apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

kgryte committed:
docs: revert description change

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after clean-up.
